temporal-sdk/src/main/java/io/temporal/internal/sync/SyncWorkflowContext.java

```diff
@@ -356,6 +356,11 @@ private ExecuteActivityParameters constructExecuteActivityParameters(
                 ProtobufTimeUtils.toProtoDuration(options.getScheduleToCloseTimeout()))
             .setHeartbeatTimeout(ProtobufTimeUtils.toProtoDuration(options.getHeartbeatTimeout()));
 
+    // FIXME: Make it possible for user to opt out of eager activity execution, eg. based on worker
```

Comment:
This is too much logic for this method. This method is pretty much brainless data transferred. This decision should be a bit up in the method structure and come here as an option/flag.

Reply:
Is your comment specifically about the FIXME, or is it about the equality check on task queue name?